front/components/assistant/conversation/ActionValidationProvider.tsx

```diff
@@ -42,38 +42,41 @@ function useValidationQueue() {
   const [currentValidation, setCurrentValidation] =
     useState<MCPActionValidationRequest | null>(null);
 
-  // Queue stores the pending validation requests
-  // The current validation request is the one being processed
-  // The queue does not stores the current validation request
-  const addToQueue = (validationRequest: MCPActionValidationRequest) => {
-    setCurrentValidation((current) => {
-      if (current === null) {
-        return validationRequest;
-      }
+  // The current validation request is the one being processed.
+  // The queue does not stores the current validation request.
+  // It's memoized since it's used as a dependency of the hook.
+  const handleValidationRequest = useCallback(
+    (validationRequest: MCPActionValidationRequest) => {
+      setCurrentValidation((current) => {
+        if (current === null) {
+          return validationRequest;
+        }
 
-      setValidationQueue((prevQueue) => [...prevQueue, validationRequest]);
-      return current;
-    });
-  };
+        setValidationQueue((prevQueue) => [...prevQueue, validationRequest]);
+        return current;
+      });
+    },
+    []
+  );
 
+  // We don't update the current validation here to avoid content flickering.
   const takeNextFromQueue = () => {
```

Comment:
Like you `handleValidationRequest` above, using `useCallback` is a nice touch

Reply:
I think this provider structure can be better yes, and let me schedule a meeting with you so that I don't keep postponing 🙈

Reply:
@johnoppenheimer I'm gonna merge this for now but feel free to correct me if I say something wrong 😄